fastcrypto/src/secp256k1.rs

```diff
@@ -422,14 +460,38 @@ impl From<Secp256k1PrivateKey> for Secp256k1KeyPair {
 }
 
 impl Secp256k1Signature {
-    /// Recover public key from signature
+    /// Return true if the public key is recoverable from this signature.
+    pub fn is_recoverable(&self) -> bool {
+        match self.sig {
+            RECOVERABLE(_) => true,
+            NONRECOVERABLE(_) => false,
+        }
```

Comment:
self.sig == RECOVERABLE(_)

Reply:
That's apparently not valid rust code. But the matches macro can do it.